app/js/lib/foxbox/network.js

```diff
@@ -172,49 +175,70 @@ export default class Network {
   }
 
   /**
-   * Ping the box to detect whether we connect locally or remotely.
+   * Ping the box to detect whether we connect locally or remotely. Since
+   * 'online' state depends on two factors: local and remote server
+   * availability, there is a slight chance that this method will cause two
+   * events e.g. if previously box was available only locally and now it's
+   * available only remotely we'll likely generate event indicating that box
+   * went offline following by another event indicating that box is online
+   * again.
    *
    * @private
    */
   [p.pingBox]() {
-    const promises = [];
-    promises.push(this[p.ping](`${this.localOrigin}/ping`)
-      .then(() => {
-        this[p.local] = true;
-      })
-      .catch(() => {
-        this[p.local] = false;
-      }));
+    const previousState = this[p.local] || this[p.remote];
+
+    this[p.ping](`${this.localOrigin}/ping`)
+      .then((isOnline) => this[p.onPong](previousState, p.local, isOnline));
 
     // @todo Find a better way to detect if a tunnel connection is active.
     if (this[p.settings].tunnelOrigin) {
-      promises.push(this[p.ping](`${this.tunnelOrigin}/ping`)
-        .then(() => {
-          this[p.remote] = true;
-        })
-        .catch(() => {
-          this[p.remote] = false;
-        }));
+      this[p.ping](`${this.tunnelOrigin}/ping`)
+        .then((isOnline) => this[p.onPong](previousState, p.remote, isOnline));
     }
-
-    Promise.all(promises).then(() => {
-      this[p.pingCallback](this[p.local] || this[p.remote]);
-    });
   }
 
   /**
-   * Get a URL but don't process the response.
+   * Performs HTTP 'GET' request to the specified URL. Returns 'true' if
+   * response was successful (any of 200-299 status codes) or 'false'
+   * otherwise.
    *
    * @param {string} url The URL to send the request to.
-   * @return {Promise}
+   * @return {Promise<boolean>}
```

Comment:
Looks like ```boolean``` is what we need from this method (and logged error if any) right now. If in the future we have rich ping response we can reconsider this.